database/src/main/java/com/firebase/ui/database/OnIndexMismatchListener.java

```diff
@@ -0,0 +1,7 @@
+package com.firebase.ui.database;
+
+import com.google.firebase.database.DataSnapshot;
+
+interface OnIndexMismatchListener {
```

Comment:
@puf the second snippet is occurs when `dataRef` returns a null snapshot. The original purpose of this PR was to change that piece of code from a log (`Log.w(TAG, "Key not found at ref: " + snapshot.getRef());`) to a callback. Is it still okay to follow through with `mJoinResolver.onJoinFailed(index, snapshot);`?